packages/components/src/entities/SampleListingPage.tsx

```diff
@@ -0,0 +1,549 @@
+                    </a>
+                    &nbsp;now or
+                </>
+            )}
+            {transactionAuditId && (
+                <>
+                    <a
+                        onClick={() =>
+                            selectGridIdsFromTransactionId(
+                                sampleListingGridId,
+                                SchemaQuery.create(SCHEMAS.SAMPLE_SETS.SCHEMA, sampleType),
+                                transactionAuditId,
+                                SAMPLES_KEY,
+                                actions
+                            )
+                        }
+                    >
+                        &nbsp;{showAddToStorage ? 'select' : 'Select'} {itThem} in the grid&nbsp;
+                    </a>
+                </>
+            )}
+            to work with {itThem}.
+        </>
+    );
+};
+
+interface OwnProps {
+    menu: ProductMenuModel;
+    menuInit: (invalidate?: boolean) => void;
+    navigate: (url: string | AppURL, replace?: boolean) => void;
+}
+
+interface BodyProps {
+    sampleListModelId: string;
+}
+
+type Props = OwnProps & BodyProps & InjectedQueryModels & WithRouterProps & InjectedRouteLeaveProps;
+
+const SampleListingPageBody: FC<Props> = props => {
+    const { menuInit, navigate, queryModels, actions, sampleListModelId, menu, setIsDirty, getIsDirty } = props;
+    const { sampleType } = props.params;
+    const { canPrintLabels, labelTemplate, printServiceUrl } = useLabelPrintingContext();
+    const { createNotification, dismissNotifications } = useNotificationsContext();
+    const {
+        detailRenderer,
+        downloadTemplateExcludeColumns,
+        getSamplesEditableGridProps,
+        AddSamplesToStorageModalComponent,
+        SampleGridButtonComponent,
+    } = useSampleTypeAppContext();
+    const [showPrintDialog, setShowPrintDialog] = useState(false);
+    const [showAddToStorage, setShowAddToStorage] = useState(false);
+    const [showConfirmDeleteSampleSet, setShowConfirmDeleteSampleSet] = useState(false);
+    const [sharedContainerPermissions, setSharedContainerPermissions] = useState<List<string>>();
+    const containerFilter = useMemo(() => getContainerFilterForLookups(), []);
+
+    const detailsModel = queryModels[DETAIL_GRID_ID];
+    const listModel = queryModels[sampleListModelId];
+    const { error, isLoaded, user } = useContainerUser(detailsModel?.getRowValue('Folder/Path'));
+
+    useEffect(() => {
+        getUserSharedContainerPermissions()
+            .then(permissions => {
+                setSharedContainerPermissions(fromJS(permissions));
+            })
+            .catch(reason => {
+                console.error(reason);
+            });
+    }, []);
+
+    useEffect(() => {
+        if (!isLoaded) return;
+
+        const { transactionAuditId, createdSampleCount, importedSampleCount } = props.location?.query;
+        const sampleSet = listModel.schemaQuery.getQuery();
```

Comment:
sampleType?

Reply:
ah yes, I was going to go back and make those sampleSet -> sampleType text changes after getting the dev complete. Thank you for the reminder.